include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -392,9 +392,11 @@ struct __gen_set_mask
         auto __val_a = __set_a[__id];
 
         auto __res = oneapi::dpl::__internal::__pstl_lower_bound(__set_b, std::size_t{0}, __nb, __val_a, __comp);
+        constexpr bool __is_difference = std::is_same_v<_SetTag, oneapi::dpl::unseq_backend::_DifferenceTag>;
```

Comment:
The construction `constexpr bool __is_difference = ` used twice in the code.
May be make sense to create some trait?

Reply:
ok